src/main/scala/streamit/client/ApacheKafkaClient.scala

```diff
@@ -0,0 +1,257 @@
+package streamit.client
+
+import java.time.{ Instant, ZoneId, ZonedDateTime, Duration => JavaDuration }
+import java.util.{ Properties, Collection => JavaCollection }
+
+import cats.effect._
+import cats.syntax.apply._
+import fs2.Stream
+import fs2.kafka.{ KafkaConsumer => _ }
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig
+import log.effect.LogWriter
+import org.apache.kafka.clients.admin.{ AdminClient, AdminClientConfig }
+import org.apache.kafka.clients.consumer._
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.serialization.Deserializer
+import org.apache.zookeeper.ZooKeeper
+import org.slf4j.LoggerFactory
+import streamit._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+object ApacheKafkaClient {
+  def apply[F[_]: ConcurrentEffect: LogWriter](kafkaSettings: KafkaSettings): ApacheKafkaClient[F] =
+    new ApacheKafkaClient(kafkaSettings)
+}
+
+/**
+  * While ideally an fs2-first client such as the fs2-kafka client would be the only one in use by the
+  * KafkaTaskRunner, it is quite limited (especially in documentation), so when we need to do lower level
+  * consumergroup manipulations, we need to resort to the standard java client, so that's all being
+  * isolated to this client.
+  */
+class ApacheKafkaClient[F[_]: ConcurrentEffect](settings: KafkaSettings)(
+  implicit logger: LogWriter[F]
+) {
+
+  // WARNING: Here be dragons.  Or at least, sphincter-tightening jumps between FP and non-FP with
+  // ugly java conversions. All help at making this less ugly, very, very welcome.
+
+  // kafka API listeners etc aren't in F[_] so we can't use 'logger' :(
+  private[this] val slfLogger = LoggerFactory.getLogger("ApacheKafkaClient")
```

Comment:
An option could be isolating the side effecting methods in `kafka-clients` behind a mini DSL parametric in `F[_]`. Anyway it's better to try it in a separate change as I'm not sure how long the tail of that is going to be.

Reply:
Yeah, I'm reluctant to end up back in this part of the code for this PR :)  But yes, this sounds great.